### PR TITLE
Guard against wildcard Imports

### PR DESCRIPTION
**Pull request by MondeAnna**
Advise against the use of wildcard imports by providing an example thereof

Related issues: clean code for Python

## Description:

We, recruits, consistently make wildcard imports when wishing to avoid having to specify exact imports. It may be wise to advise against such an act at a level higher than individual reviews.

## I solemnly swear that:

- [ ] I ran the hugo server and looked at my changed in the browser with my own eyes
- [ ] I ran the linter and there were no errors
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Review by ry-oc — APPROVED** [review]

File: content/topics/clean-code/python/_index.md [hunk 1]
```diff
@@ -92,6 +92,16 @@ It's okay to say this though:
 from subprocess import Popen, PIPE
 ```
 
+Guard against wildcard imports
+
```
Comment (Rokhuda) [inline content/topics/clean-code/python/_index.md]: Perhaps add a little context as to why wildcard imports should be avoided.

**Review by Rokhuda — CHANGES_REQUESTED** [review]

**Comment by MondeAnna** [comment]
### Morning

Context added ^_^

**Review by Rokhuda — CHANGES_REQUESTED** [review]
May I suggest you just add a link that provides details about wildcards?

File: content/topics/clean-code/python/_index.md [hunk 2]
```diff
@@ -119,9 +123,44 @@ If this spelling causes local name clashes, then spell them explicitly:
...
+from statistics import *
+
+# Yes
+from math import cos, sin, tan
+from statistics import mean, median, mode
+```
+
+The primary reason for ensuring that imports are explicit is to prevent the cluttering of the namespace, else one of the following is likely to happen:
+- You have an outright naming conflict between your imports, for instance `maths.sqrt` and `statistics.sqrt`
+- Imports are sequential, thus the final wildcard's classes and functions will override whateter came before. Should this be the incorrect class or function, you will struggle to figure out why the code does not function as intended.
```
Comment (Rokhuda) [inline content/topics/clean-code/python/_index.md]: Please use grammarly, there is a few grammar issues.

**Review by Rokhuda — CHANGES_REQUESTED** [review]

File: content/topics/clean-code/python/_index.md [hunk 3]
```diff
@@ -119,9 +123,44 @@ If this spelling causes local name clashes, then spell them explicitly:
...
+if __name__ == "__main__":
+    print(dt.tzinfo())
+    print(datetime.tzinfo())
+```
+
+For further details follow these links:
+- [Wildcard imports should not be used](https://rules.sonarsource.com/python/RSPEC-2208)
+- [Why You Should Avoid Wildcard Imports](https://www.youtube.com/shorts/A9AhKxSBZ0Q)
+
+In summary, wildcard imports should be avoided, as they make it unclear which names are present in the namespace, confusing both readers and many automated tools. There is one defensible use case for a wildcard import, which is to republish an internal interface as part of a public API (for example, overwriting a pure Python implementation of an interface with the definitions from an optional accelerator module and exactly which definitions will be overwritten isn't known in advance).
```
Comment (Rokhuda) [inline content/topics/clean-code/python/_index.md]: It is very hard to visualize what you referring to in these examples. The Links do give better insight.

File: content/topics/clean-code/python/_index.md [hunk 4]
```diff
@@ -119,9 +123,44 @@ If this spelling causes local name clashes, then spell them explicitly:
...
+- You have an outright naming conflict between your imports, for instance `maths.sqrt` and `statistics.sqrt`
+- Imports are sequential, thus the final wildcard's classes and functions will override whatever came before. Should this be the incorrect class or function, you will struggle to figure out why the code does not function as intended.
+- Ensures that the code is in keeping with the _Clean Code_ mantra as each import just _*"does one thing"*_, better still other users of the code can know at a glance which aspects of the import are meant to be used.
+- This also guards against changes that get made to the imported module (such as an upgrade to Python or a third-party library), all of which could cause the above-mentioned to occur down the line, long after the code has been merged.
 
-When republishing names this way, the guidelines below regarding public and internal interfaces still apply.
+Should there still be a need to import multiple aspects of a module or a module in its entirety, attempt to import submodules, ensuring unique names and dot call whatever classes or functions you desire
```
Comment (Rokhuda) [inline content/topics/clean-code/python/_index.md]: This info is extrapolated in the links provided. Should we just point them to the links for the rest of the info?

File: content/topics/clean-code/python/_index.md [hunk 5]
```diff
@@ -119,9 +123,44 @@ If this spelling causes local name clashes, then spell them explicitly:
...
+
+```
+# No:
+from math import *
+from statistics import *
+
+# Yes
+from math import cos, sin, tan
+from statistics import mean, median, mode
+```
```
Comment (Rokhuda) [inline content/topics/clean-code/python/_index.md:132]: This can be kept

**Review by Rokhuda — CHANGES_REQUESTED** [review]

**Review by TheAbstract — APPROVED** [review]
Looks good 👍🏾.

**Pull request merged**